frame/ethereum/src/lib.rs

```diff
@@ -162,6 +164,15 @@ decl_module! {
 					)
 				),
 			);
+			// move block hash pruning window by one block
+			let block_hash_count = T::BlockHashCount::get();
+			let to_remove = n.saturating_sub(block_hash_count).saturating_sub(One::one());
+			// keep genesis hash
+			if !to_remove.is_zero() {
+				BlockHash::remove(U256::from(
+					UniqueSaturatedInto::<u32>::unique_saturated_into(to_remove)
+				));
+			}
```

Comment:
Can this go in `on_initialize`? I think it's preferable to use `on_initialize` when possible because `on_initialize` is responsible for accounting all the weight of both `on_initialize` and `on_finalize`. So having the code there is more transparent.

Reply:
Just followed what `frame_system` does, although we could do it in either of them I think.